test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,393 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id: test_filters.cpp 7683 2012-10-23 02:49:03Z rusu $
+ *
+ */
+
+#include <pcl/test/gtest.h>
+
+#include <random>
+
+#include <pcl/point_types.h>
+#include <pcl/surface/convex_hull.h>
+#include <pcl/common/common.h>
+#include <pcl/filters/crop_hull.h>
+
+
+namespace
+{
+  template <class PointGenerator1, class PointGenerator2>
+  void complexTest(
+      pcl::CropHull<pcl::PointXYZ> cropHullFilter,
+      PointGenerator1 insidePointGenerator,
+      PointGenerator2 outsidePointGenerator)
+  {
+    pcl::PointCloud<pcl::PointXYZ>::Ptr insideCubeCloud(new pcl::PointCloud<pcl::PointXYZ>);
+    pcl::PointCloud<pcl::PointXYZ>::Ptr outsideCubeCloud(new pcl::PointCloud<pcl::PointXYZ>);
+    pcl::PointCloud<pcl::PointXYZ>::Ptr mixedCubeCloud(new pcl::PointCloud<pcl::PointXYZ>);
+    size_t const chunkSize = 100;
+    {
+      for (size_t i = 0; i < chunkSize; ++i)
+      {
+        insideCubeCloud->push_back(insidePointGenerator());
+        outsideCubeCloud->push_back(outsidePointGenerator());
+      }
+      *mixedCubeCloud = (*insideCubeCloud) + (*outsideCubeCloud);
+    }
+    std::vector<int> emptyIndices;
+    std::vector<int> lowerIndices(insideCubeCloud->size());
+    std::vector<int> upperIndices(insideCubeCloud->size());
+    std::iota(lowerIndices.begin(), lowerIndices.end(), 0);
+    std::iota(upperIndices.begin(), upperIndices.end(), lowerIndices.size());
+
+    struct CheckEquals {
+      pcl::CropHull<pcl::PointXYZ> cropHullFilter;
+      CheckEquals(pcl::CropHull<pcl::PointXYZ> const & cropHullFilter)
+        : cropHullFilter(cropHullFilter)
+      {}
+      void check(
+          std::vector<int> const & expectedFilteredIndices,
+          pcl::PointCloud<pcl::PointXYZ>::ConstPtr inputCloud)
+      {
+        std::vector<bool> expectedFilteredMask(inputCloud->size(), false);
+        std::vector<int> expectedRemovedIndices;
+        pcl::PointCloud<pcl::PointXYZ> expectedCloud;
+        pcl::copyPointCloud(*inputCloud, expectedFilteredIndices, expectedCloud);
+        for (int idx : expectedFilteredIndices) {
+          expectedFilteredMask[idx] = true;
+        }
+        for (size_t i = 0; i < inputCloud->size(); ++i) {
+          if (!expectedFilteredMask[i]) {
+            expectedRemovedIndices.push_back(i);
+          }
+        }
+
+        cropHullFilter.setInputCloud(inputCloud);
+
+        std::vector<int> filteredIndices;
+        cropHullFilter.filter(filteredIndices);
+        ASSERT_EQ (expectedFilteredIndices.size(), filteredIndices.size());
```

Comment:
PCL has `EXPECT_EQ_VECTORS` and other goodies in `pcl/pcl_tests.h` for these situations

Reply:
applied the suggestion